### ISSUE TITLE
PlaidLink.open never returns

### ISSUE THREAD

**Issue by almovsesanso** [jorgefspereira/plaid_flutter#110]
Thus any `await PlaidLinks.open()` calls will never finish, regardless of success or fail.

**Comment by GeorgePagounis** [comment]
This is happening to me as well, only on ios, in web `await PlaidLinks.open()` returns after the initial configuration is done, and can call callbacks after

**Comment by almovsesanso** [comment]
> This is happening to me as well, only on ios, in web `await PlaidLinks.open()` returns after the initial configuration is done, and can call callbacks after

It is happening to me on Android as well.

**Comment by palmsey** [comment]
You can use the `onSuccess` stream of `PlaidLink` to listen for the callback success.

For example:

```
PlaidLink.open(configuration: configuration);
await for (final result in PlaidLink.onSuccess) {
  return result.publicToken;
}
```

Thanks to #108 that helped me figure this out.

**Comment by GeorgePagounis** [comment]
@palmsey this is different. This is what is called after the user enters successfully his credentials and authorized with Plaid. We want the state of after the plaid link is successfully initialized (which never returns on ios and android, but does on the web). Since it is a different outcome per device I think this is a bug.

**Comment by amitrotner** [comment]
I managed to fix this on Android by sending the `android_package_name` field in the [Create Link Token](https://plaid.com/docs/api/tokens/#linktokencreate) request.

**Comment by almovsesanso** [comment]
@jorgefspereira Which version of the package has or will have the fix? I tried looking at the commit history and wasn't sure if I could find when/if the fix was checked in.

**Comment by jorgefspereira** [comment]
> @jorgefspereira Which version of the package has or will have the fix? I tried looking at the commit history and wasn't sure if I could find when/if the fix was checked in.

You are right. I got confused with another ticket. Fixed on 3.1.4
